### PR TITLE
"param ** value" optimization for params that are positive powers of 2" fixed

### PR DESCRIPTION
**Pull request by Ldelaney**
See the "param ** value" optimization PR ( #4331 )

This is clean branch to revert-the-revert (PR #4352)  of PR #4331.  The problems causing that revert to be needed have been fixed. 

I had missed that there was already a param**value overload for when both integers are unsigned, which caused some ambiguity errors. Also added a check for a zero or negative exponent.

### CODE REVIEW

**Comment by Ldelaney** [comment]
@psahabu Would you be willing to review again?

File: modules/internal/ChapelBase.chpl [hunk 1]
```diff
@@ -363,11 +363,31 @@ module ChapelBase {
...
+  inline proc _expBaseHelp(param a: int, b) {
+    if b == 0 then 
+      return 1:a.type;
+    if b < 0 then
+        return 0; // same behavior as signed int **
+    var c = 0;
+    var x: int = a;
+    while (x > 1) // shift right to count the power
+    {
+      c = c + 1;
```
Comment (psahabu) [inline modules/internal/ChapelBase.chpl]: You could do c += 1 here.

**Comment by psahabu** [comment]
Otherwise, looks good to me.

File: modules/internal/ChapelBase.chpl (hunk 1 above)
Comment (Ldelaney) [inline modules/internal/ChapelBase.chpl]: I can indeed

**Comment by Ldelaney** [comment]
pending the CI build I think this is ready to merge.

**Pull request merged**